smoke/internal/host/util.go

```diff
@@ -0,0 +1,298 @@
+	require.NoError(t, err, "should have found builds to check after generating tasks")
+	require.NotEmpty(t, builds, "must have at least one build to test after generating tasks")
+
+	// Isolate the generated tasks from the original tasks.
+	_, generatedTasks := utility.StringSliceSymmetricDifference(originalTasks, builds[0].Tasks)
+	require.NotEmpty(t, generatedTasks, "should have generated at least one task")
+
+	// Check that the generated tasks eventually finish and check their output.
+	internal.CheckTaskStatusAndLogs(ctx, t, params.APIParams, client, agent.HostMode, generatedTasks)
+}
+
+// triggerRepotracker makes a request to the Evergreen app server's REST API to
+// run the repotracker. This is a necessary prerequisite to catch the smoke
+// test's mainline commits up to the latest version. That way, when the smoke
+// test submits a manual patch, the total diff size against the base commit is
+// not so large.
+// Note that this returning success means that the repotracker will run
+// eventually. It does *not* guarantee that it has already run, nor that it has
+// actually managed to pick up the latest commits from GitHub.
+func triggerRepotracker(ctx context.Context, t *testing.T, params SmokeTestParams, client *http.Client) {
+	grip.Info("Attempting to trigger repotracker to run.")
+
+	const repotrackerAttempts = 5
+	for i := 0; i < repotrackerAttempts; i++ {
+		time.Sleep(2 * time.Second)
+		grip.Infof("Requesting repotracker for evergreen project. (%d/%d)", i+1, repotrackerAttempts)
+		_, err := internal.MakeSmokeRequest(ctx, params.APIParams, http.MethodPost, client, fmt.Sprintf("/rest/v2/projects/%s/repotracker", params.ProjectID))
+		if err != nil {
+			grip.Error(errors.Wrap(err, "requesting repotracker to run"))
+			continue
+		}
+
+		grip.Info("Successfully triggered repotracker to run.")
+
+		return
+	}
+
+	require.FailNow(t, "ran out of attempts to trigger repotracker", "could not successfully trigger repotracker after %d attempts", repotrackerAttempts)
+}
+
+// waitForRepotracker waits for the repotracker to pick up new commits and
+// create versions for them. The particular versions that it creates for these
+// commits is not that important, only that they exist.
+func waitForRepotracker(ctx context.Context, t *testing.T, params SmokeTestParams, client *http.Client) {
+	grip.Info("Waiting for repotracker to pick up new commits.")
+
+	const repotrackerAttempts = 10
+	for i := 0; i < repotrackerAttempts; i++ {
+		time.Sleep(2 * time.Second)
+		respBody, err := internal.MakeSmokeRequest(ctx, params.APIParams, http.MethodGet, client, fmt.Sprintf("/rest/v2/projects/%s/versions?limit=1", params.ProjectID))
+		if err != nil {
+			grip.Error(errors.Wrapf(err, "requesting latest version for project '%s'", params.ProjectID))
+			continue
+		}
+		if len(respBody) == 0 {
+			grip.Error(errors.Errorf("did not find any latest revisions yet for project '%s'", params.ProjectID))
```

Comment:
Can we update this (and the following) to grip.Errorf as well?

Reply:
Sure, I updated the others in the smoke tests.